src/Microsoft.VisualStudio.SDK.Analyzers/VSSDK013ThreadAffinitizedMEFConstruction.cs

```diff
@@ -0,0 +1,185 @@
+
+        /// <summary>
+        /// A reusable descriptor for diagnostics produced by this analyzer.
+        /// </summary>
+        internal static readonly DiagnosticDescriptor Descriptor = new DiagnosticDescriptor(
+            id: Id,
+            title: "Avoid UI thread in MEF Part construction",
+            messageFormat: "MEF part construction must not have UI thread affinity",
+            helpLinkUri: Utils.GetHelpLink(Id),
+            category: "Reliability",
+            defaultSeverity: DiagnosticSeverity.Warning,
+            isEnabledByDefault: true);
+
+        /// <inheritdoc />
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Descriptor);
+
+        private ImmutableArray<TypeMatchSpec> MembersRequiringMainThread { get; set; }
+
+        /// <inheritdoc />
+        public override void Initialize(AnalysisContext context)
+        {
+            if (context is null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze);
+
+            // Register for compilation first so that we only activate the analyzer for applicable compilations
+            context.RegisterCompilationStartAction(startCompilation =>
+            {
+                this.MembersRequiringMainThread = AdditionalFilesHelpers.GetMembersRequiringMainThread(startCompilation.Options, startCompilation.CancellationToken);
+                INamedTypeSymbol? importingConstructorAttribute = startCompilation.Compilation.GetTypeByMetadataName(Types.ImportingConstructorAttribute.FullName)?.OriginalDefinition;
+                INamedTypeSymbol? partImportsSatisfiedNotificationInterface = startCompilation.Compilation.GetTypeByMetadataName(Types.IPartImportsSatisfiedNotification.FullName)?.OriginalDefinition;
+                INamedTypeSymbol? exportAttributeType = startCompilation.Compilation.GetTypeByMetadataName(Types.ExportAttribute.FullName)?.OriginalDefinition;
+
+                var operationKinds = ImmutableArray.Create<OperationKind>(
+                    OperationKind.MethodReference,
+                    OperationKind.InstanceReference,
+                    OperationKind.FieldReference,
+                    OperationKind.ObjectOrCollectionInitializer,
+                    OperationKind.Invocation, // Method calls
+                    OperationKind.MemberInitializer, // For static member access
+                    OperationKind.PropertyReference, // For property access
+                    OperationKind.ObjectCreation); // For object creation scenarios
+
+                startCompilation.RegisterOperationAction(
+                    Utils.DebuggableWrapper(c => this.AnalyzeOperation(c, exportAttributeType, importingConstructorAttribute, partImportsSatisfiedNotificationInterface)),
+                    operationKinds);
```

Comment:
this takes params-OperationKinds.  So you could just pass the above list here directly.